### PR TITLE
chore(ui): adding support for Datadog Data Store

### PR DESCRIPTION
**Pull request by danielbdias**
This PR adds support for Datadog as trace data store. The configuration page follows the same model as the Lightstep and New Relic integrations:

image

## Changes

- API Protos: add Datadog as a type of data source
- UI: add configuration page for Datadog
- Docs: added first documentation page on `Configuration -> Connecting to Data Stores -> Datadog`

## Fixes

- #1970 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: examples/quick-start-datadog-nodejs/test-api.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ type: Test
 spec:
   id: W656Q0c4g
   name: http://app:8080
-  description: akadlkasjdf
+  description: Calling Hello World API
```
Comment (xoscar) [inline examples/quick-start-datadog-nodejs/test-api.yaml:5]: What was the problem with `akadlkasjdf`?

**Review by xoscar — APPROVED** [review]
Clean and simple, GJ @danielbdias!

**Review by jorgeepc — CHANGES_REQUESTED** [review]
Hey @danielbdias looks great. I think we need to add some support in the server code too. Please take a look at
- https://github.com/kubeshop/tracetest/blob/main/server/tracedb/tracedb.go#L76
- https://github.com/kubeshop/tracetest/blob/main/server/model/data_stores.go#L85

**Comment by xoscar** [comment]
@jorgeepc nice catch, I thought there was a BE PR checking that scenario!

File: examples/quick-start-datadog-nodejs/test-api.yaml (hunk 1 above)
Comment (danielbdias) [reply examples/quick-start-datadog-nodejs/test-api.yaml:5]: My documentation OCD won't free me if if don't update this description 😂

File: docs/docs/configuration/connecting-to-data-stores/datadog.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+### CLI
+
+Or, if you prefer using the CLI, you can use this file config.
+
+```yaml
+type: DataStore
+spec:
+  name: OpenTelemetry Collector pipeline
+  type: otlp
```
Comment (jorgeepc) [inline docs/docs/configuration/connecting-to-data-stores/datadog.md:74]: should the type be `datadog` here? I know it's going to work with otlp, but to be able to display the proper selected option in the UI should we have `datadog` here?

**Review by jorgeepc — APPROVED** [review]

File: docs/docs/configuration/connecting-to-data-stores/datadog.md (hunk 2 above)
Comment (danielbdias) [reply docs/docs/configuration/connecting-to-data-stores/datadog.md:74]: I believe that we can change this for Datadog. I'll do this on the next PR, that handles examples and documentation.

**Pull request merged**